### ISSUE TITLE
Vertical nav active state not available to screen readers

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3576]
## Describe the bug

A white background for the active link is used to convey information that is not communicated to screen reader users. 

## How to reproduce

https://v1.clarity.design/vertical-nav/nav-groups/credit

- Using a screen reader, go to https://v1.clarity.design/vertical-nav/nav-groups/credit
- Read any of vertical nav items that has a background color of #fff
- Notice that the screen reader does not announce the meaning of the #fff background

## Expected behavior

Dermine what is meant by the #fff and communicate it textually. aria-elected and aria-current are potential properties that could be used. If these don't match the intent of the background color I recommend using aria-label to communicate the meaning of the #fff background.

If the intent of the this background image is to convey the active state of the link I recommend using the active CSS psuedo-class seen below.

a:active { }

Review the definition of the active pseudo-class from https://www.w3.org/TR/CSS2/selector.html#dynamic-pseudo-classes below and decide if we are trying to communicate the active state of the link with the #fff background color.

The :active pseudo-class applies while an element is being activated by the user. For example, between the times the user presses the mouse button and releases it.

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

_Add any other notes about the problem here._

**Comment by andreiconstantinbytex** [comment]
Tried to investigate this but the following website cannot be found: https://v1.clarity.design/vertical-nav/nav-groups/credit
![image](https://user-images.githubusercontent.com/52167120/63944491-468d5880-ca7a-11e9-8b7d-466893b95a7c.png)

**Comment by Jinnie** [comment]
Hi @andreiconstantinbytex 
Just go here:
https://v1.clarity.design/vertical-nav/basic-structure/charmander#examples
and then navigate to Vertical Nav Groups

I'll spawn a new issue for the broken navigation.

**Comment by Jinnie** [comment]
Here's the navigation issue: #3776

**Comment by bdryanovski** [comment]
I when over this and it seems that both `aria-selected` and `aria-current` had no effect on the announcement. Maybe it is only JAWS (didn't test it) or another screen reader. On macOS, I expected to get something as `visited, link, current page|step, <link innerText>`. I found this testing [page](http://design-patterns.tink.uk/aria-current/) and even there I didn't get any of expected responses. So I'm thinking that these tags are not supported or I'm doing it wrong.

So I went with `aria-label`. The same code is valid for the `ClrVerticalNav` - the reason that this code is duplicated is that the navigation and nested navigation are handled by two components - examples below will include only `ClrVerticalNavGroupChildren` for simplicity.

```ts
import { Component, Renderer2, ContentChildren } from '@angular/core';
import { RouterLinkActive } from '@angular/router';

@Component({
  selector: 'clr-vertical-nav-group-children',
  template: `
        <ng-content></ng-content>
    `,
})
export class ClrVerticalNavGroupChildren {
  constructor(public renderer : Renderer2) {}

  @ContentChildren(RouterLinkActive) routerLinks;

  ngAfterContentChecked(): void {
    if (this.routerLinks.length) {
      this.routerLinks.map((link) => {
        if (link.isActive) {
          this.renderer.setAttribute(link.element.nativeElement, 'aria-label', 'active');
        } else {
          this.renderer.removeAttribute(link.element.nativeElement, 'aria-label');
        }
      });
    }
  }
}
```

`aria-label` is replacing the navigation content and only returning the content that I'm providing - so I'm losing useful information like for example the navigation text:

<img width="600" alt="Screen Shot 2019-12-02 at 11 31 50 AM" src="https://user-images.githubusercontent.com/204564/69948342-24cbc380-14f8-11ea-82f4-6eabce7e5f2e.png">

So to fetch the actual content I will have to search for `.nav-text` class inside and retrieve it from there. And I think this is going too far with all the additional `renderrer` calls for adding and removing attributes.

I gonna try to find a more easy way of accessing which navigation node is active at the moment, and play a little with it. I will need some time on that.

PS.
I notice that with `innerText/contentText` on grouped elements I'm getting all the nodes as a single string so this could be tricky to extract. 

<img width="536" alt="Screen Shot 2019-12-02 at 12 33 33 PM" src="https://user-images.githubusercontent.com/204564/69952496-fb169a80-14ff-11ea-88e9-a599a97dbc21.png">
 
```ts
const firstElement = (el.innerText.split("\n"))[0]
```

**Comment by gnomeontherun** [comment]
@chlane Can you help understand what the right aria- implementation is for this use case?

@bdryanovski If we can't use aria-selected/current, then we can talk about the label options.

**Comment by chlane** [comment]
Assuming that the background color is meant to convey the current page, we can use aria-current="page". aria-current="page" works for all browser/screen reader combinations other than JAWS 2019/Edge. Be sure to remove the aria-label that was added to convey this state as it could override the actual link text.  See my test results for aria-current="page" at http://cx-guidelines.vmware.com/examples/ariaCurrent.html.

**Comment by bdryanovski** [comment]
If we want to use the `aria-current` my previous example above will work fine (instead of `aria-label`).

Are we making any changes to the color of the `selected` navigation item? It seems that there is at least one request for that, I'm sure that there was one more but could not find it now.

I could make a PR for aria-current for now .. ?
